app/api/PostReview.js

```diff
@@ -9,12 +9,50 @@ export default async (req, res) => {
   const queryParams = [reviewer, rating_one, rating_two, rating_three, text, course_number, quarter, professor];
   
   // Execute the query
-  pool.query(query, queryParams, (err, results) => {
+  await pool.query(query, queryParams, (err, results) => {
     if (err) {
       console.error('Error executing MySQL query:', err);
       res.status(500).json({ error: 'Internal Server Error' });
     } else {
       res.json({ success: true, message: 'Data inserted successfully' });
     }
   });
+
+  // SQL query to calculate the average ratings from all the reviews for the course
+  const avgQuery = `
+    SELECT 
+      AVG(rating_one) AS avg_rating_one, 
+      AVG(rating_two) AS avg_rating_two,
+      AVG(rating_three) AS avg_rating_three,
+    FROM reviews 
+    WHERE class_id = (SELECT class_id FROM courses WHERE number = ?)`;
+  const avgParams = [course_number];
+  let avgResult;
+
+  // Execute the average query  
+  try {
+    avgResult = await pool.query(avgQuery, avgParams);
+    res.json({ success: true, message: 'Averages calculated successfully' });
+  } catch (avgError) {
+    console.error('Error executing AVG query:', avgError);
+    res.status(500).json({ error: 'Error calculating averages' });
+    return; // Exit the function early to avoid sending the success response
+  }
+
+  // Extract the average rating_one value from the result
+  const { avg_rating_one, avg_rating_two, avg_rating_three } = avgResult[0];
+
+  // SQL query to update the ratings for the course
+  const updateQuery = 'UPDATE courses SET rating_one = ?, rating_two = ?, rating_three = ? WHERE class_id = ?';
```

Comment:
class_id does not correspond to class number. I think it should class_number = ?, not class_id. Also test this to make sure

Reply:
Fixed.